predicators/train_refinement_estimator.py

```diff
@@ -0,0 +1,285 @@
+                ground_nsrts.append(ground_nsrt)
+                if time.perf_counter() - ground_nsrt_start_time > \
+                        ground_nsrt_timeout:
+                    raise PlanningTimeout("Planning timed out in grounding!")
+    elif CFG.sesame_grounder == "fd_translator":
+        # WARNING: there is no easy way to check the timeout within this call,
+        # since Fast Downward's translator is a third-party function. We'll
+        # just check the timeout afterward.
+        ground_nsrts = list(
+            utils.all_ground_nsrts_fd_translator(nsrts, objects, predicates,
+                                                 types, init_atoms, task.goal))
+        if time.perf_counter() - ground_nsrt_start_time > ground_nsrt_timeout:
+            raise PlanningTimeout("Planning timed out in grounding!")
+    else:
+        raise ValueError(
+            f"Unrecognized sesame_grounder: {CFG.sesame_grounder}")
+    metrics: Metrics = defaultdict(float)
+    # Optionally exclude NSRTs with empty effects, because they can slow
+    # the search significantly, so we may want to exclude them.
+    nonempty_ground_nsrts = [
+        nsrt for nsrt in ground_nsrts
+        if (nsrt.add_effects | nsrt.delete_effects)
+    ]
+    all_reachable_atoms = utils.get_reachable_atoms(nonempty_ground_nsrts,
+                                                    init_atoms)
+    if not task.goal.issubset(all_reachable_atoms):
+        raise PlanningFailure(f"Goal {task.goal} not dr-reachable")
+    reachable_nsrts = [
+        nsrt for nsrt in nonempty_ground_nsrts
+        if nsrt.preconditions.issubset(all_reachable_atoms)
+    ]
+    heuristic = utils.create_task_planning_heuristic(
+        CFG.sesame_task_planning_heuristic, init_atoms, task.goal,
+        reachable_nsrts, predicates, objects)
+    try:
+        gen = _skeleton_generator(
+            task, reachable_nsrts, init_atoms, heuristic, seed,
+            CFG.refinement_data_skeleton_generator_timeout, metrics,
+            CFG.refinement_data_num_skeletons)
+        for skeleton, atoms_sequence in gen:
+            necessary_atoms_seq = utils.compute_necessary_atoms_seq(
+                skeleton, atoms_sequence, task.goal)
+            refinement_start_time = time.perf_counter()
+            _, suc = run_low_level_search(
+                task, option_model, skeleton, necessary_atoms_seq, seed,
+                CFG.refinement_data_low_level_search_timeout, metrics,
+                CFG.horizon)
+            # Calculate time taken for refinement.
+            refinement_time = time.perf_counter() - refinement_start_time
+            # Add datapoint to dataset
+            data.append((
+                task.init,
+                skeleton,
+                atoms_sequence,
+                suc,
+                refinement_time,
+            ))
+    except _MaxSkeletonsFailure:
+        # Done finding skeletons
+        return
+
+
+def _get_data_file_path() -> Path:
```

Comment:
in our main code, we just enforce the standard naming scheme. what's the reason to allow overrides?